File: orion-frontend/src/types/comunidad.ts
/** Comunidad del catálogo que administra el admin (para el combobox de pedidos). */
export interface Comunidad {
  id: number;
  nombre: string;
  /** Código que el admin comparte con la comunidad para registrar pedidos. */
  codigo?: string;
  activo: boolean;
}

/** Payload para crear/editar una comunidad. */
export interface ComunidadInput {
  nombre: string;
  codigo?: string;
}

File: orion-backend/api/src/main/java/com/orionlogistic/api/comunidades/Comunidad.java
package com.orionlogistic.api.comunidades;

import jakarta.persistence.*;
import lombok.*;

/** Comunidad del catálogo (tabla `comunidades`) para el combobox de pedidos. */
@Entity
@Table(name = "comunidades")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Comunidad {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 100)
    private String nombre;

    /** Código que el admin comparte con la comunidad; el cliente lo ingresa al registrar. */
    @Column(length = 50)
    private String codigo;

    @Column(nullable = false)
    @Builder.Default
    private Boolean activo = true;
}
